src/plugins/intel_cpu/src/emitters/jit_snippets_emitters.cpp

```diff
@@ -719,6 +721,291 @@ void StoreConvertEmitter::emit_isa(const std::vector<size_t> &in, const std::vec
+                auto N_ = n ? N0_tail : N0 - N0_tail;
+                auto K_ = k ? K0_tail : K0 - K0_tail;
+                auto beta = k && brgCtxs0[getBrgIdx(m, 0, n)].K != 0 ? 1.0f : 0.0f;
+
+                brgemmCtx.M = M_;
+                brgemmCtx.N = N_;
+                brgemmCtx.K = K_;
+                brgemmCtx.LDA = leading_dimensions[0];
+                brgemmCtx.LDB = leading_dimensions[1];
+                brgemmCtx.LDC = leading_dimensions[2];
+                brgemmCtx.dt_in0 = static_cast<dnnl_data_type_t>(DnnlExtensionUtils::IEPrecisionToDataType(brg0Prc));
+                brgemmCtx.dt_in1 = static_cast<dnnl_data_type_t>(DnnlExtensionUtils::IEPrecisionToDataType(brg1Prc));
+                brgemmCtx.beta = beta;
+
+                // don't create brgemm kernels for empty tiles
+                if (M_ != 0 && K_ != 0 && N_ != 0) {
+                    if (brg0BaseIdx == -1)
+                        brg0BaseIdx = getBrgIdx(m, k, n);
+                    initBrgemm(brgemmCtx, brgKernels0[getBrgIdx(m, k, n)], brg0WithAMX);
+                }
+            }
+        }
+    }
+}
+
+void BrgemmEmitter::initBrgemm(brgemmCtx& ctx, std::unique_ptr<brgemm_kernel_t>& brgKernel, bool use_amx) const {
+    brgemm_t brgDesc;
+    brgemm_strides_t strides {static_cast<dnnl_dim_t>(ctx.M * ctx.K), static_cast<dnnl_dim_t>(ctx.K * ctx.N)};
+    // When implementing int8 support, note that isa logics is more complicated in the MHA node
+    auto status = brgemm_desc_init(&brgDesc, host_isa_, brgemm_strd, ctx.dt_in0, ctx.dt_in1,
+                                   false, false, brgemm_row_major, 1.f, ctx.beta, ctx.LDA, ctx.LDB, ctx.LDC, ctx.M, ctx.N, ctx.K, &strides);
+    if (status != dnnl_success)
+        IE_THROW() << "BrgemmEmitter cannot initialize brgemm descriptor due to invalid params";
 
+    ctx.is_with_amx = use_amx;
+    status = brgemm_init_tiles(brgDesc, ctx.palette);
+    if (use_amx)
+        amx_tile_configure(ctx.palette);
+
+    ctx.is_with_comp = ctx.dt_in0 == dnnl_data_type_t::dnnl_s8 && !ctx.is_with_amx;
+
+    brgemm_kernel_t* brgKernel_ = nullptr;
+    status = brgemm_kernel_create(&brgKernel_, brgDesc);
+    if (status != dnnl_success)
+        IE_THROW() << "BrgemmEmitter cannot create brgemm kernel due to invalid params";
+    brgKernel.reset(brgKernel_);
+}
+
+void BrgemmEmitter::emit_impl(const std::vector<size_t>& in,
+                              const std::vector<size_t>& out,
+                              const std::vector<size_t>& pool,
+                              const std::vector<size_t>& gpr,
+                              const ov::intel_cpu::emitter_context *emit_context) const {
+    if (host_isa_ == cpu::x64::sse41) {
+        emit_isa<cpu::x64::sse41>(in, out);
+    } else if (host_isa_ == cpu::x64::avx2) {
+        emit_isa<cpu::x64::avx2>(in, out);
+    } else if (host_isa_ == cpu::x64::avx512_core) {
+        emit_isa<cpu::x64::avx512_core>(in, out);
```

Comment:
brgemm block form oneDNN supports at least avx512_core, so need to throw exception for sse and avx2

Reply:
Good point, thank you